### ISSUE TITLE
clang/libgcrypt v1.9.3 issue  (CentOS 8 workflow)

### ISSUE THREAD

**Issue by maxirmx** [rnpgp/rnp#1668]
**Target:**  CentOS 8
**Tests**   (normal, 2.3.1, clang, clang++)
**Module:**  libgcrypt / 1.9.3
**Error:**
```
basic.c:315:16: error: inline assembly requires more registers than available
  asm volatile("movdqu %[data0], %%xmm0\n"
               ^
basic.c:315:16: error: inline assembly requires more registers than available
```

Other tests use  libgcrypt /v1.8.7  and  libgcrypt / v1.9.0-beta277  which are compiled without issues.

_Originally posted by @maxirmx in https://github.com/rnpgp/rnp/issues/1129#issuecomment-959510564_

**Comment by ronaldtse** [comment]
@maxirmx we don't need GnuPG to work with clang, only RNP libraries. Thanks.

**Comment by maxirmx** [comment]
same fix as #1669 (force gcc toolchain for GnuPg)
